dotnet/src/Connectors/Connectors.Ollama/Settings/OllamaPromptExecutionSettings.cs

```diff
@@ -111,6 +112,22 @@ public float? Temperature
         }
     }
 
+    /// <inheritdoc/>
+    public override FunctionChoiceBehavior? FunctionChoiceBehavior
+    {
+        get => base.FunctionChoiceBehavior;
+        set
+        {
+            if (value is not null && value.GetType().Name != "AutoFunctionChoiceBehavior")
+            {
+                throw new NotSupportedException(
+    "Currently, Ollama does only supports 'Auto' choice behavior. " +
+    "See Ollama docs at https://github.com/ollama/ollama/blob/55ea963/docs/openai.md#supported-request-fields to see whether support has since been added.");
+            }
+            base.FunctionChoiceBehavior = value;
+        }
+    }
```

Comment:
We could allow this to fail at runtime and avoid having to track when this get's added.

I'm not sure we want to track every permutation of FunctionChoiceBehavior across all connectors and check what is and is not supported.

Reply:
@markwallace-microsoft I used this perspective to allow the fail fast approach.

Currently we only have 3 permutations abstracted, leaving this to fail only in the runtime I personally think is a bad experience for the user if he can find faster it can't be used.

Reply:
We are only doing it for this one Connector, we have other connectors which don't support function calling and we don't have this behaviour. 
If we are going to do this we should do it consistently across all connectors and ideally centralise the logic.

Reply:
AS a off line agreement we thought would be better fail at runtime instead of introducing a new pattern.